### PR TITLE
add multi-stage pipeline using gatsby

### PR DESCRIPTION
**Pull request by ferantivero**
- add multi-stage pipeline (ci/cd)
  - remove manual static website upload steps
  - build the SPA static content from ci stage in the YAML pipeline 
  - upload build artifact to  Azure Storage static website hosting from cd stage in the YAML pipeline

solved: #42

### CODE REVIEW

**Comment by ferantivero** [comment]
cc/ @kirpasingh

File: src/ClientApp/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        AZURE_CLIENT_ID=$(azureClientId)
+        AZURE_API_CLIENT_ID=$(azureApiClientId)
+        AZURE_API_URL=$(azureApiUrl)
+        EOF
+      displayName: 'configure azure tenant, client and api details'
+
+    - script: |
+        cd src/ClientApp
+        npm install
+        npx gatsby build
```
Comment (fsimonazzi) [inline src/ClientApp/azure-pipelines.yml]: do we need to install the cli if running with npx?

File: src/ClientApp/readme.md [hunk 2]
```diff
@@ -75,7 +123,7 @@ az storage account create --name $STORAGE_ACCOUNT_NAME --resource-group $RESOURC
 az storage blob service-properties update --account-name $STORAGE_ACCOUNT_NAME --static-website --404-document 404.html --index-document index.html
 
 # Upload the web site
-az storage blob upload-batch --source ./ClientApp/public --destination \$web --account-name $STORAGE_ACCOUNT_NAME
+az storage blob upload-batch --source ./ClientApp/drop --destination \$web --account-name $STORAGE_ACCOUNT_NAME
```
Comment (fsimonazzi) [inline src/ClientApp/readme.md]: won't this be done through the pipeline now? If an early smoke test, I'd move it out of the mandatory storage account set up instructions.

File: src/ClientApp/readme.md [hunk 3]
```diff
@@ -48,19 +59,56 @@ sed "s#<application id>#$CLIENT_APP_ID#g" | \
 sed "s#<api application id>#$API_APP_ID#g" | \
 sed "s#<URL of the API Management API endpoint>#${APIM_GATEWAY_URL}#g" \
 > config.js.tmp && mv config.js.tmp src/ClientApp/src/services/config.js
```
Comment (fsimonazzi) [inline src/ClientApp/readme.md]: I'd expect this setup to be removed and the variables to be pushed to the pipeline's config.

File: src/ClientApp/readme.md (hunk 2 above)
Comment (fsimonazzi) [reply src/ClientApp/readme.md]: See I was reviewing an out of date copy. Resolving

File: src/ClientApp/readme.md (hunk 3 above)
Comment (fsimonazzi) [reply src/ClientApp/readme.md]: See I was reviewing an out of date copy. Resolving

File: src/ClientApp/azure-pipelines.yml (hunk 1 above)
Comment (ferantivero) [reply src/ClientApp/azure-pipelines.yml]: good catch, please let me test and remove accordingly.

Comment (ferantivero) [reply src/ClientApp/azure-pipelines.yml]: addressed from https://github.com/mspnp/serverless-reference-implementation/pull/43/commits/28b14afa0689a73b8806682b898c562a6f164c0c

File: src/ClientApp/readme.md [hunk 4]
```diff
@@ -1,8 +1,9 @@
 # Serverless client app
 
-## Prerequisite
+## Prerequisites
 
- - Install [node](https://nodejs.org/en/download/)
+ - [Azure CLI 2.0.49 or later](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
```
Comment (fsimonazzi) [inline src/ClientApp/readme.md]: some of the instructions require a newer version of the Azure CLI, at least 2.0.69

Comment (ferantivero) [reply src/ClientApp/readme.md]: addressed https://github.com/mspnp/serverless-reference-implementation/pull/43/commits/dd6c114b0269d6523b252911c2a267e145ec1144

**Pull request merged**